op-e2e/actions/proofs/isthmus_fork_test.go

```diff
@@ -85,35 +102,39 @@ func TestIsthmusActivationAtGenesis(gt *testing.T) {
 	got, ok := genesisPayload.CheckBlockHash()
 	require.Equal(t, got, reproduced.Hash())
 	require.True(t, ok, "CheckBlockHash must pass")
-}
 
-// There are 2 stages pre-Isthmus that we need to test:
-// 1. Pre-Canyon: withdrawals root should be nil
-// 2. Post-Canyon: withdrawals root should be EmptyWithdrawalsHash
-func TestWithdrawlsRootPreCanyonAndIsthmus(gt *testing.T) {
-	t := helpers.NewDefaultTesting(gt)
-	dp := e2eutils.MakeDeployParams(t, helpers.DefaultRollupTestParams())
-	canyonOffset := hexutil.Uint64(2)
+	safeBlock := env.Sequencer.L2Safe()
+	require.NoError(t, err, "error fetching latest block")
 
-	log := testlog.Logger(t, log.LvlDebug)
-
-	dp.DeployConfig.L1CancunTimeOffset = &canyonOffset
+	env.RunFaultProofProgram(t, safeBlock.Number, testCfg.CheckResult, testCfg.InputParams...)
+}
 
-	// Activate pre-canyon forks at genesis, and schedule Canyon the block after
-	dp.DeployConfig.ActivateForkAtOffset(rollup.Canyon, uint64(canyonOffset))
-	require.NoError(t, dp.DeployConfig.Check(log), "must have valid config")
+// Test_ProgramAction_IsthmusWithdrawlsRoot tests the withdrawals root in the header:
+// - pre canyon
+// - post canyon but pre Isthmus
+// - post Isthmus
+// It does this by activating the relevant forks at genesis.
```

Comment:
This is a slight change to how the test worked pre migration.

Then, it only ran assertions pre canyon and post canyon, and canyon was activated after genesis. 

Now, it runs the whole test three times and always activates the forks at genesis.

Reply:
Did that here [4fdbf1e](https://github.com/ethereum-optimism/optimism/pull/15296/commits/4fdbf1e63aa115ded79a8ebb41fa93b2d8083b26).